Modules/Magnetometer/magnetometer.py

```diff
@@ -0,0 +1,29 @@
+from .. import config
+
+if not config.HAS_HARDWARE:
+    from . import mpu9250
+else:
+    import random
+
+class magnetometer:
+    def __init__(self):
+        if not config.HAS_HARDWARE:
+            self.sensor = mpu9250.mpu9250()
+
+    # Reading data from magnetometer
+    # NOT MULTI THREADING SAFE
+    # use magnetometerMT for that
+    def __get_x__(self):
+        return str(self.__get_axes__()[0])
+
+    def __get_y__(self):
+        return str(self.__get_axes__()[1])
+
+    def __get_z__(self):
+        return str(self.__get_axes__()[2])
+
+    def __get_axes__(self):
+        if not config.HAS_HARDWARE:
+            return self.sensor.read_xyz()
+        else:
+            return (random.uniform(0,1),random.uniform(0,1),random.uniform(0,1))        
```

Comment:
I guess this line is not meant to be merged into `master`, right?

Reply:
There is a config variable called HAS_HARDWARE that should be set to false if the code is being run on a developer's machine, for example when debugging the communications module, the magnetometer would still give us data (random data but data) and won't crash because the module is not connected.

Reply:
To be honest, I was dazzled by the random line and I did not notice the `HAS_HARDWARE` configuration variable. (Whoops?)

I think it's descriptive enough. But as for `not config.HAS_HARDWARE`, I would return a `(0,) * 3` and not a random tuple.